src/index.js

```diff
@@ -1,29 +1,40 @@
 import React from "react";
 import ReactDOM from "react-dom";
-import { BrowserRouter, Route } from "react-router-dom";
-import { Security, LoginCallback } from "@okta/okta-react";
+import { BrowserRouter as Router, Route, useHistory } from "react-router-dom";
+import { Security, LoginCallback, SecureRoute } from "@okta/okta-react";
 
-import ExampleList from "./components/ExampleListComponent"
+import ExampleList from "./components/ExampleListComponent";
 import Login from "./components/LoginComponent";
 import Home from "./components/HomeComponent";
 import { config } from "./utils/oktaConfig";
 
 ReactDOM.render(
-  <BrowserRouter>
+  <Router>
     <React.StrictMode>
-      <Security {...config}>
-        <Route path="/" exact>
-          <Home />
-        </Route>
-        <Route path="/login">
-          <Login />
-        </Route>
-        <Route path="/implicit/callback" component={LoginCallback} />
-        <Route path="/items">
-          <ExampleList />
-        </Route>
-      </Security>
+      <App />
     </React.StrictMode>
-  </BrowserRouter>,
+  </Router>,
   document.getElementById("root")
 );
+
+function App() {
+  // The reason to declare App this way is so that we can use any helper functions we'd need for business logic, in our case auth.
+  // React Router has a nifty useHistory hook we can use at this level to ensure we have security around our routes.
+  const history = useHistory();
+
+  const authHandler = () => {
+    // We pass this to our <Security /> component that wraps our routes.
+    // It'll automatically check if userToken is available and push back to login if not :)
+    history.push("/login");
+  };
```

Comment:
It may be useful to abstract this away into a HOC in future to check auth and redirect to a specified route at the individual route level. Worth exploring in our next version. 👍

Reply:
Love it! Yeah I think that throwing this in Context or a HOC is a great idea.